.github/workflows/changelogs/changelogs.py

```diff
@@ -0,0 +1,639 @@
+            out += " for package " + self.package
+        if self.file:
+            out += " in file " + self.file
+            if self.line:
+                out += "#L" + str(self.line)
+                if self.end_line:
+                    out += "-" + str(self.end_line)
+        return out
+
+class ChangelogValidator:
+    """Class that handles the changelog validation
+
+    Validates the changelog entries by enforcing various rules, such as;
+      - Basic typos, spacing, capitalization and line length issues
+      - Ensuring the changelogs are added for correct packages
+
+    Additionally, the following checks are made if a tracker file is
+    provided;
+      - Correct spelling of tracker IDs defined by IBS/OBS
+      - Correct matching of tracker IDs between the PR title, comment
+        messages and the changelog entries (only if a valid PR number is
+        provided)
+      - For Bugzilla trackers, the bug exists in the specified Bugzilla
+        host
+      - For Bugzilla trackers, the bug is reported for the "SUSE Manager"
+        product (passes with a warning if the check fails)
+
+    If a tracker file and a PR number is provided, the following environment
+    variables must be also set:
+
+        GH_TOKEN: A GitHub access token with the basic privileges
+        BZ_TOKEN: A valid Bugzilla API key
+
+    The trackers XML file (trackers.xml):
+        This is an XML document that contains definitions and formats of all
+        the various types of tracker IDs recognized by IBS/OBS. The document is
+        provided by the OBS/IBS APIs.
+
+        It can be obtained by sending a request to the '/issue-trackers'
+        endpoint of the OBS/IBS API:
+
+        `osc api /issue-trackers`
+    """
+
+    def __init__(self, spacewalk_root: str, git_repo: str, pr_number: int, max_line_length: int,
+                 regex_rules: type[RegexRules]):
+        if pr_number and not os.getenv("GH_TOKEN"):
+            raise Exception("GitHub API key not set. Please set it in 'GH_TOKEN' environment variable.")
+
+        self.spacewalk_root = spacewalk_root
+        self.git_repo = git_repo
+        self.pr_number = pr_number
+        self.max_line_length = max_line_length
+        self.regex = regex_rules
+        if regex_rules.trackers:
+            self.bzapi = self.get_bugzilla_api()
+
+    def get_bugzilla_api(self) -> type[bugzilla.Bugzilla]:
+        """Initialize and authenticate the Bugzilla API"""
+
+        api_key = os.getenv("BZ_TOKEN")
+        if not api_key:
+            raise Exception("Bugzilla API key not set. Please set it in 'BZ_TOKEN' environment variable.")
+
+        uri = os.getenv("BUGZILLA_URI", DEFAULT_BUGZILLA_URI)
+        try:
+            logging.info(f"Initializing Bugzilla API at '{uri}'")
+            bzapi = bugzilla.Bugzilla(uri, api_key=api_key)
+        except requests.exceptions.ConnectionError as e:
+            raise ConnectionError(f"Cannot connect to the Bugzilla API at '{uri}'")
```

Comment:
`ConnectionError` is not a known name here:
```suggestion
            raise requests.exceptions.ConnectionError(f"Cannot connect to the Bugzilla API at '{uri}'")
```
If that's too long for your taste (it is for mine :D), we could update the import:
```
from requests import exceptions as req_exceptions
```
and use `req_exceptions.ConnectionError` here

Reply:
IIUC, python also has a built-in `ConnectionError` (not related to `requests`). That's what I meant to use here.